pkg/fakerp/store/type.go

```diff
@@ -0,0 +1,85 @@
+package store
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+)
+
+type Store interface {
+	Put(value *api.OpenShiftManagedCluster) error
+	Get() (*api.OpenShiftManagedCluster, error)
+	UpdateState(state api.ProvisioningState) error
+	Delete() error
+}
+
+type Storage struct {
+	storageDir string
+	sync.Mutex
+	log *logrus.Entry
+}
+
+var _ Store = &Storage{}
+
+var containerServices = "containerservice.yaml"
+
+func NewStore(log *logrus.Entry, storageDir string) Storage {
+	return Storage{
+		storageDir: storageDir,
+		log:        log,
+	}
+}
+
+func (s *Storage) Put(cs *api.OpenShiftManagedCluster) error {
+	s.Lock()
+	defer s.Unlock()
+	s.log.Debugf("storage: Put %s/%s", s.storageDir, containerServices)
+	bytes, err := yaml.Marshal(cs)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(fmt.Sprintf("%s/%s", s.storageDir, containerServices), bytes, 0600)
+}
+
+func (s *Storage) Get() (*api.OpenShiftManagedCluster, error) {
```

Comment:
I suspect this will need to relax to return a nil cs and no error if the file doesn't exist

Reply:
Did a slightly different implementation of `store`.